src/main.rs

```diff
@@ -7,19 +7,37 @@ extern crate cortex_m;
 #[macro_use(entry, exception)]
 extern crate cortex_m_rt as rt;
 
+#[cfg(feature = "semihosting")]
+extern crate cortex_m_semihosting as semihosting;
+
 extern crate panic_abort;
 
 use core::arch::arm;
 
 use rt::ExceptionFrame;
 
+#[cfg(feature = "semihosting")]
+use core::fmt::Write;
+
 entry!(main);
 
 fn main() -> ! {
     let x = 42;
 
     loop {
         unsafe { arm::__NOP() }
+
+        #[cfg(feature = "semihosting")]
+        unsafe {
+            let mut hstdout = semihosting::hio::hstdout().unwrap();
+            write!(hstdout, "x = {}\n", x);
+
+            // system call to exit successfully
+            semihosting::syscall1(
```

Comment:
Could you also send a PR to `cortex-m-semihosting` adding an `fn exit() -> !` function that does this syscall? `syscall` is not divergent so you'll need to add a call [`unreachable`](https://crates.io/crates/unreachable) after the syscall.

Reply:
`cortex-m-semihosting` already has `exit()` function which I couldn't find before.  I just use this!